src/LibraryManager.Contracts/FileHelpers.cs

```diff
@@ -245,21 +253,33 @@ public static bool DeleteFiles(IEnumerable<string> filePaths)
         /// Deletes folders if empty
         /// </summary>
         /// <param name="folderPaths"></param>
+        /// <param name="rootDirectory"></param>
         /// <returns></returns>
-        public static bool DeleteEmptyFoldersFromDisk(IEnumerable<string> folderPaths)
+        public static bool DeleteEmptyFoldersFromDisk(IEnumerable<string> folderPaths, string rootDirectory)
         {
+            if (folderPaths.Count() == 0)
+            {
+                return true;
+            }
+
+            HashSet<string> newFolderPaths = new HashSet<string>();
+
             try
             {
                 foreach (string path in folderPaths)
                 {
-                    DirectoryInfo directory = new DirectoryInfo(path);
-                    if (directory.Exists && IsDirectoryEmpty(path))
+                    if (!path.Equals(rootDirectory, StringComparison.CurrentCultureIgnoreCase))
```

Comment:
This block also needs to make sure that the directories being deleted are inside the `rootDirectory`

Reply:
Use the full path here to compare.

---
In reply to: [200728323](https://github.com/aspnet/LibraryManager/pull/286#discussion_r200728323) [](ancestors = 200728323)

Reply:
This should be done using `LocalPathComparer`, to handle the cases we discussed for the comparer.



---
In reply to: [200867103](https://github.com/aspnet/LibraryManager/pull/286#discussion_r200867103) [](ancestors = 200867103,200728323)